mypy/server/deps.py

```diff
@@ -368,9 +368,17 @@ def visit_del_stmt(self, o: DelStmt) -> None:
 
     # Expressions
 
-    # TODO
-    #   dependency on __init__ (e.g. ClassName())
-    #   super()
+    def process_global_ref_expr(self, o: RefExpr) -> None:
+        if o.fullname is not None:
+            self.add_dependency(make_trigger(o.fullname))
+
+        # If this is a reference to a type, generate a dependency to its
+        # constructor. This will be spurious in cases like isinstance
+        # calls and except statements, but that's no big deal.
```

Comment:
What about class attribute references? If things like `Foobar.method()` depend on `FooBar.__init__`, this could have performance implications.